test/metabase/public_settings/premium_features_test.clj

```diff
@@ -254,3 +257,19 @@
   (testing "If premium-embedding-token is nil, the token-status setting should also be nil."
     (mt/with-temporary-setting-values [premium-embedding-token nil]
       (is (nil? (premium-features/token-status))))))
+
+(deftest active-users-count-setting-test
+  (t2.with-temp/with-temp
+    [User _ {}
+     User _ {}
+     User _ {}]
+    ;; premium-features/active-users-count  is cached so it could be cause flaky test
+    ;; rebinding to avoid caching
+    (testing "returns the number of active uesrs"
+      (with-redefs [premium-features/cached-active-user-count #'premium-features/active-users-count*]
+        (is (= (t2/count :core_user :is_active true)
+               (premium-features/active-users-count)))))
```

Comment:
it's a bummer to have to make the expected a DB function instead of, you know, `3`. I wish we had a better way of clearing out values in the DB for testing. I wrote `with-single-admin-user` for similar reasons, but of course that in particular doesn't help here. I suppose the most generic form of it would be something like

```clojure
(with-only User
  [_ {}]
  [_ {}]
  [_ {:is_active false}]
  [{admin-id :id} {is_admin true}]
...)
```
sorta thing.

No need to write it for this PR, but something I'm going to mull over

Reply:
I initially use the `metabase-enterprise.serialization.test-util/with-empty-h2-app-db` macro. but since it's in an enterprise ns and I have to make lots of changes to make it available in oss ns due to cyclic deps.

If you think it's worth it I'm all in for it.
<img width="658" alt="Screenshot 2023-03-01 at 18 03 02" src="https://user-images.githubusercontent.com/25661381/222121494-fb5e3ddb-1087-4f74-8560-38a371fc8bd1.png">

Reply:
^ I tried to move it into `metabase.test` so we could use it more widely